pkg/controller/operators/operatorcondition_controller.go

```diff
@@ -252,6 +253,27 @@ func (r *OperatorConditionReconciler) ensureDeploymentEnvVars(operatorCondition
 	return nil
 }
 
+func (r *OperatorConditionReconciler) syncOperatorConditionStatus(operatorCondition *operatorsv2.OperatorCondition) error {
+	r.log.V(4).Info("Sync operatorcondition status")
+	currentGen := operatorCondition.ObjectMeta.GetGeneration()
+	changed := false
+	for _, cond := range operatorCondition.Spec.Conditions {
+		if c := meta.FindStatusCondition(operatorCondition.Status.Conditions, cond.Type); c != nil {
+			if cond.Status == c.Status && c.ObservedGeneration == currentGen {
```

Comment:
Why do we check that `cond.Status == c.Status` here?

Reply:
This is a strict comparison that will storm out any possible manipulation the status in the object.Status while maintaining the same ObservedGeneration. We can relax this by only comparing ObservedGeneration. That's totally fine. That will allow an escape hatch if something happens with OperationCondition that causes the generation out of sync.